### PR TITLE
fix uid overflow on 32bit platforms

### PR DESCRIPTION
**Pull request by petrosagg**
Linux uses u32 as the {u,g}id map structures[1] but golang incorrectly uses ints[2]. Since changing the standard library is hard this commit only changes the parsing code to use the correct types and truncates the result to MaxInt.

The effect is that on 32bit platforms the namespaces will have at most 2 billion available uid instead of 4 billion.

[1] https://github.com/torvalds/linux/blob/v4.17/include/linux/user_namespace.h#L17-L21
[2] https://github.com/golang/go/blob/5c11480631f5654e9e6937ff08c453660138c64d/src/syscall/exec_linux.go#L14-L20

### CODE REVIEW

**Comment by petrosagg** [comment]
Friendly ping for this PR :)

**Comment by julz** [comment]
Hi @petrosagg - sorry for slow progress getting to this one, I pulled it up our backlog just now, you can see our (again, slow, sorry!) progress getting to it here: https://www.pivotaltracker.com/story/show/159823655. It's currently 2 from the top.

File: max_valid_uid.go [hunk 1]
```diff
@@ -25,18 +27,18 @@ func (u IDMap) MaxValid() (int, error) {
...
 	scanner := bufio.NewScanner(f)
 	for scanner.Scan() {
-		var container, host, size int
+		var container, host, size uint
 		if _, err := fmt.Sscanf(scanner.Text(), "%d %d %d", &container, &host, &size); err != nil {
 			return 0, ParseError{Line: scanner.Text(), Err: err}
 		}
 
-		m = Max(m, container+size-1)
+		m = minUint(container+size-1, uint(maxInt))
```
[the remark this answers is not in the document]
Comment (petrosagg) [reply max_valid_uid.go]: fixed the maxUint omission

**Comment by BooleanCat** [comment]
LGTM - we'll bump this in garden-runc-release too

**Pull request merged**